ql/src/java/org/apache/hadoop/hive/ql/parse/repl/dump/events/AbstractEventHandler.java

```diff
@@ -94,7 +94,8 @@ protected void writeFileEntry(String dbName, Table table, String file, BufferedW
           throws IOException, LoginException, MetaException, HiveFatalException {
     HiveConf hiveConf = withinContext.hiveConf;
     String distCpDoAsUser = hiveConf.getVar(HiveConf.ConfVars.HIVE_DISTCP_DOAS_USER);
-    if (!Utils.shouldDumpMetaDataOnly(table, withinContext.hiveConf)) {
+    if (!Utils.shouldDumpMetaDataOnlyForExternalTables(table, withinContext.hiveConf)
```

Comment:
for external table we don't dump data related events ..so i don't think we need to add this check